### PR TITLE
Make atomic_write work on paths with no directory separator

### LINKED ISSUE #19: Cannot use atomicwrites on filename with no directory component
When running the example in the README with atomicwrites 1.1.0:

```
from atomicwrites import atomic_write

with atomic_write('foo.txt', overwrite=True) as f:
    f.write('Hello world.')
```

I get:

```
Traceback (most recent call last):
  File "/Users/ryan/temp/atomictest.py", line 4, in <module>
    f.write('Hello world.')
  File "/Users/ryan/.pyenv/versions/3.5.2/lib/python3.5/contextlib.py", line 66, in __exit__
    next(self.gen)
  File "/Users/ryan/.pyenv/versions/3.5.2/lib/python3.5/site-packages/atomicwrites/__init__.py", line 152, in _open
    self.commit(f)
  File "/Users/ryan/.pyenv/versions/3.5.2/lib/python3.5/site-packages/atomicwrites/__init__.py", line 177, in commit
    replace_atomic(f.name, self._path)
  File "/Users/ryan/.pyenv/versions/3.5.2/lib/python3.5/site-packages/atomicwrites/__init__.py", line 89, in replace_atomic
    return _replace_atomic(src, dst)
  File "/Users/ryan/.pyenv/versions/3.5.2/lib/python3.5/site-packages/atomicwrites/__init__.py", line 46, in _replace_atomic
    _sync_directory(os.path.dirname(dst))
  File "/Users/ryan/.pyenv/versions/3.5.2/lib/python3.5/site-packages/atomicwrites/__init__.py", line 38, in _sync_directory
    fd = os.open(directory, 0)
FileNotFoundError: [Errno 2] No such file or directory: ''
```

The reason is that `os.path.dirname('foo.txt')` returns `''` instead of `'.'`.

It looks like a bug was introduced here: https://github.com/untitaker/python-atomicwrites/commit/4c7de64e07240e91482a845afc122e3c6aff0bb1#diff-530c1a5f2cffcd30669f3bb3ebbafc75R31

### PR DESCRIPTION
**Pull request by DarwinAwardWinner**
Fixes #19, and adds a test for it.

(I wasn't able to figure out how to actually run the tests, so you'll have to check that my test actually works.)

### CODE REVIEW

**Comment by untitaker** [comment]
Hello and thanks for this PR -- I think if we apply `abspath` before calling `dirname` this problem should solve itself.

**Comment by DarwinAwardWinner** [comment]
Does that potentially introduce a race condition where the filesystem could change between the call to abspath and the creation of the file?

**Comment by untitaker** [comment]
Using `strace python` it seems that `os.path.abspath` calls `getcwd`. This might be enough for a race condition. Maybe `os.path.normpath` then?

**Comment by DarwinAwardWinner** [comment]
Looks like normpath works:

``` python
In [4]: os.path.normpath("")
Out[4]: '.'
```

 I'll update the PR.

**Comment by untitaker** [comment]
I added a contributing.rst that should explain how to run the tests.

File: tests/test_atomicwrites.py [hunk 1]
```diff
@@ -65,3 +65,24 @@ def test_open_reraise(tmpdir):
             f.name = "asdf"
             # Now trigger our own exception.
             assert False, "Intentional failure for testing purposes"
+
+
+def test_atomic_write_in_pwd(tmpdir):
+    curdir= os.getcwd()
```
Comment (untitaker) [inline tests/test_atomicwrites.py]: `os` is not imported.

**Comment by untitaker** [comment]
I'm restarting [the build](https://travis-ci.org/untitaker/python-atomicwrites/builds/151836293) which should fail because of the broken import.

File: tests/test_atomicwrites.py [hunk 2]
```diff
@@ -65,3 +66,24 @@ def test_open_reraise(tmpdir):
             f.name = "asdf"
             # Now trigger our own exception.
             assert False, "Intentional failure for testing purposes"
+
+
+def test_atomic_write_in_pwd(tmpdir):
+    orig_curdir= os.getcwd()
```
Comment (untitaker) [inline tests/test_atomicwrites.py]: Space missing before `=`.

Comment (DarwinAwardWinner) [inline tests/test_atomicwrites.py]: Hmm. It seems that `tox -e py-stylecheck`, which I was using as recommended by CONTRIBUTING.rst, doesn't catch this (and possibly doesn't run anything at all), while `tox -e py35-stylecheck` does. Anyway, I'll fix it.

**Comment by untitaker** [comment]
Excellent, thanks!

**Pull request merged**